### DISCUSSION TITLE
Webhook to python

### DISCUSSION THREAD

**Discussion by TeStevens315** [python-telegram-bot/python-telegram-bot#4299] [Q&A]
Hello,   

I am trying to send a telegram message (consisting of a few comma seperated values) from Google AppSheet to python via webhook to telegram.   The python script will take the coma seperated values and make them into a list that I will send to my label printer.   I have parts of it working: when I trigger the webhook message shows up in telegram, and my python script can also interact with telegram. 

My issue is when I type the coma seperated values into telegram my bot proceses them fine and returns the list.    However when I send the same commands from my webhook my bot never sees them.    If I copy and paste the command in telegram then my bot will see it and process it.

Here is a screenshot of telegram.    The first message (8:27) is from the message from my webhook, the next message was me copying and pasting the message back into telegram and the last two messages are the responses from my python bot.

![image](https://github.com/python-telegram-bot/python-telegram-bot/assets/20211492/35983c9d-676a-4e99-8f26-e129a9674582)

Thanks for any ideas on how to make this work.   

**Here is my webhookurl:** `https://api.telegram.org/bot<bottolken>/sendMessage`  **webhook body:**
```python
{ "chat_id":<chatid>, "text":"<<CONCATENATE( 
"/printme"
," "
 ,[Sample ID]
, ","
,[Date Created]
, ","
,[Name]
 )>>", "parse_mode":"MarkdownV2" }
```

**My Python Below:**  

```python
"""
Simple Bot to reply to Telegram messages.

First, a few handler functions are defined. Then, those functions are passed to
the Application and registered at their respective places.
Then, the bot is started and runs until we press Ctrl-C on the command line.

Usage:
Basic Echobot example, repeats messages.
Press Ctrl-C on the command line or send a signal to the process to stop the
bot.
"""

import logging

from telegram import ForceReply, Update
from telegram.ext import Application, CommandHandler, ContextTypes, MessageHandler, filters, CallbackContext

import requests
TOKEN = "<bottolken>"
#chat_id = "<chatid>"





#SEND MESSAGE
message = "hello from your telegram bot"
url = f"https://api.telegram.org/bot{TOKEN}/sendMessage?chat_id={chat_id}&text={message}"



# Enable logging
logging.basicConfig(
    format="%(asctime)s - %(name)s - %(levelname)s - %(message)s", level=logging.INFO
)
# set higher logging level for httpx to avoid all GET and POST requests being logged
logging.getLogger("httpx").setLevel(logging.WARNING)

logger = logging.getLogger(__name__)


# Define a few command handlers. These usually take the two arguments update and
# context.
async def start(update: Update, context: ContextTypes.DEFAULT_TYPE) -> None:
    """Send a message when the command /start is issued."""
    #print(requests.get(url).json()) # this sends the message
    user = update.effective_user

    await update.message.reply_html(
        rf"Hi {user.mention_html()}!",
        reply_markup=ForceReply(selective=True),
    )


async def help_command(update: Update, context: ContextTypes.DEFAULT_TYPE) -> None:
    """Send a message when the command /help is issued."""
    await update.message.reply_text("Help!")


async def echo(update: Update, context: ContextTypes.DEFAULT_TYPE) -> None:
    """Echo the user message."""
    message = "echobot: "
    url = f"https://api.telegram.org/bot{TOKEN}/sendMessage?chat_id={chat_id}&text={message}"
    print(requests.get(url).json()) # this sends the message
    await update.message.reply_text(update.message.text)



# async def printme(update: Update, context: CallbackContext) -> None:
async def printme(update: Update, context: ContextTypes.DEFAULT_TYPE):
    """Send a message with the arguments passed by the user, when the command /chart is issued."""
    input_mex = update.message.text
    input_args = input_mex.split('/printme ')[1]
    print(requests.get(url).json()) # this sends the message
    update.message.reply_text(input_args)
    splitarges = input_args.split(',')
    await update.message.reply_text(splitarges)


def main() -> None:
    """Start the bot."""
    # Create the Application and pass it your bot's token.
    application = Application.builder().token(TOKEN).build()

    # on different commands - answer in Telegram
    application.add_handler(CommandHandler("start", start))
    application.add_handler(CommandHandler("help", help_command))
    application.add_handler(CommandHandler('printme', printme))

    # on non command i.e message - echo the message on Telegram
    application.add_handler(MessageHandler(filters.TEXT & ~filters.COMMAND, echo))
    # Run the bot until the user presses Ctrl-C
    application.run_polling(allowed_updates=Update.ALL_TYPES)

    #
# Get the dispatcher to register handlers
    
    


if __name__ == "__main__":
    main()`


```

**Comment by Bibo-Joshi** [comment]
Hi.

```python
url = f"https://api.telegram.org/bot{TOKEN}/sendMessage?chat_id={chat_id}&text={message}"
print(requests.get(url).json()) # this sends the message
```

this does the exact same as 

```python
context.bot.send_message(chat_id=chat_id, text=message)
```

i.e. it sends a message *from* the bot to the `chat_id`.

Bots don't get updates about messages that they send to other users. So that way, your `CommandHandler('printme', printme)` can not be triggered.

This means two things:

1. I recommend not to make manual requests to the bot APi but to instead use `python-telegram-bot` for that - after all, that's what this library is for ;)
2. If you want your bot to handle updates via a webhook, you'll acutally have to set up a webhook and listen to updates sent to it. Please have a look at https://github.com/python-telegram-bot/python-telegram-bot/wiki/Frequently-Asked-Questions#i-want-to-handle-updates-from-an-external-service-in-addition-to-the-telegram-updates-how-do-i-do-that and the resources linked therein.

**Reply by TeStevens315** [reply]
Thanks for the quick response, I think thats exaclty the point in the right direction I needed.  

Thanks for the help.
